packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    SkiaGoldClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= SkiaGoldClient();
+
+    // Calculate the appropriate basedir for the current test context.
+    final FileSystem fs = goldens.fs;
+    final Directory testDirectory = fs.directory(defaultComparator.basedir);
+    final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
+    return FlutterSkiaGoldFileComparator(goldens.comparisonRoot.childDirectory(testDirectoryRelativePath).uri);
+  }
+
+  /// Updates the local golden file with the most recent imageBytes before
+  /// uploading to Skia Gold for comparison.
+  ///
+  /// The returned future completes with a boolean value that indicates whether
+  /// [imageBytes] passed the Skia Gold pixel test.
+  ///
+  /// In the case of the given golden file being unavailable, the comparator
+  /// will throw a [TestFailure].
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
+
+    final File goldenFile = getGoldenFile(golden);
+    if (!goldenFile.existsSync()) {
+      throw TestFailure('Could not be compared against non-existent file: "$golden"');
+    }
+    await _skiaClient.auth(fs.directory(basedir));
+    await _skiaClient.imgtestInit();
+    return await _skiaClient.imgtestAdd(golden.path, goldenFile);
+  }
+
+  /// Updates the golden file identified by [golden] with [imageBytes].
+  ///
+  /// This will be invoked in lieu of [compare] when [autoUpdateGoldenFiles]
+  /// is `true` (which gets set automatically by the test framework when the
+  /// user runs `flutter test --update-goldens`).
+  ///
+  /// The [FlutterSkiaGoldFileComparator] will always update the local golden
+  /// file so upload to Skia Gold.
   @override
   Future<void> update(Uri golden, Uint8List imageBytes) async {
-    final File goldenFile = _getGoldenFile(golden);
+    final File goldenFile = getGoldenFile(golden);
     await goldenFile.parent.create(recursive: true);
     await goldenFile.writeAsBytes(imageBytes, flush: true);
   }
 
-  File _getGoldenFile(Uri uri) {
+  /// Returns the golden file identified by the given [Uri].
+  @override
+  File getGoldenFile(Uri uri) {
     return fs.directory(basedir).childFile(fs.file(uri).path);
   }
+
+  /// Modifies the given [Uri] to incorporate the current test library as a
+  /// prefix in the golden file name.
+  @override
+  Uri getTestUri(Uri key, int version) {
+    return key;
+  }
+
+  Uri _addPrefix(Uri golden) {
+    final String prefix = basedir.pathSegments[basedir.pathSegments.length - 2];
+    return Uri.parse(prefix + '.' + golden.toString());
+  }
+}
+
+/// Utilized in the Main method to set [goldenFileComparator] to an instance of
```

Comment:
A doc comment usually doesn't need to describe where a thing is used. Describe what it does instead, e.g. "Decides based on the current environment wether golden tests should be performed agains Skia Gold" or something like that.